util/ConfigFileProcess.cpp

```diff
@@ -0,0 +1,39 @@
+#include "ConfigFileProcess.h"
+
+//
+// combining config, persistent_config and commandline args
+//
+void ConfigFileProcess(const std::vector<std::string> args)
```

Comment:
Also the function name doesn't represent what the function is doing.  `ReadOptions` would be more fitting and also in line with the naming convention.